pkg/test/environment.go

```diff
@@ -0,0 +1,152 @@
+}
+
+func NewEnvironment(ctx context.Context, env *coretest.Environment) *Environment {
+	//
+	ec2api := &fake.EC2API{}
+	ssmapi := &fake.SSMAPI{}
+
+	// cache
+	ssmCache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	ec2Cache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	kubernetesVersionCache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	instanceTypeCache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	unavailableOfferingsCache := awscache.NewUnavailableOfferings()
+	launchTemplateCache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	subnetCache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	securityGroupCache := cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval)
+	fakePricingAPI := &fake.PricingAPI{}
+
+	// Providers
+	pricingProvider := pricing.NewProvider(ctx, fakePricingAPI, ec2api, "", make(chan struct{}))
+	subnetProvider := subnet.NewProvider(ec2api, subnetCache)
+	securityGroupProvider := securitygroup.NewProvider(ec2api, securityGroupCache)
+	amiProvider := amifamily.NewProvider(env.Client, env.KubernetesInterface, ssmapi, ec2api, ssmCache, ec2Cache, kubernetesVersionCache)
+	amiResolver := amifamily.New(env.Client, amiProvider)
+	instanceTypesProvider := instancetype.NewProvider("", instanceTypeCache, ec2api, subnetProvider, unavailableOfferingsCache, pricingProvider)
+	launchTemplateProvider :=
+		launchtemplate.NewProvider(
+			ctx,
+			launchTemplateCache,
+			ec2api,
+			amiResolver,
+			securityGroupProvider,
+			ptr.String("ca-bundle"),
+			make(chan struct{}),
+			net.ParseIP("10.0.100.10"),
+			"https://test-cluster",
+		)
+	instanceProvider :=
+		instance.NewProvider(ctx,
+			"",
+			ec2api,
+			unavailableOfferingsCache,
+			instanceTypesProvider,
+			subnetProvider,
+			launchTemplateProvider,
+		)
+	cloudProvider := cloudprovider.New(ctx, instanceTypesProvider, instanceProvider, env.Client, amiProvider)
+
+	return &Environment{
+		EC2API:     ec2api,
+		SSMAPI:     ssmapi,
+		PricingAPI: fakePricingAPI,
+
+		SSMCache:                  ssmCache,
+		EC2Cache:                  ec2Cache,
+		KubernetesVersionCache:    kubernetesVersionCache,
+		InstanceTypeCache:         instanceTypeCache,
+		LaunchTemplateCache:       launchTemplateCache,
+		SubnetCache:               subnetCache,
+		SecurityGroupCache:        securityGroupCache,
+		UnavailableOfferingsCache: unavailableOfferingsCache,
+
+		InstanceTypesProvider:  instanceTypesProvider,
+		InstanceProvider:       instanceProvider,
+		SubnetProvider:         subnetProvider,
+		SecurityGroupProvider:  securityGroupProvider,
+		PricingProvider:        pricingProvider,
+		AMIProvider:            amiProvider,
+		AMIResolver:            amiResolver,
+		LaunchTemplateProvider: launchTemplateProvider,
+		CloudProvider:          cloudProvider,
+	}
+}
+
+func (env *Environment) ResetCache() {
```

Comment:
```suggestion
func (env *Environment) Reset() {
```

This isn't just resetting the cache. Also, I think a short, simpler name is better